mlx/backend/cuda/quantized.cu

```diff
@@ -0,0 +1,383 @@
+// Copyright © 2025 Apple Inc.
+
+#include "mlx/backend/cuda/device.h"
+#include "mlx/backend/cuda/kernel_utils.cuh"
+#include "mlx/backend/gpu/copy.h"
+#include "mlx/dtype_utils.h"
+#include "mlx/fast_primitives.h"
+
+#include <cooperative_groups.h>
+#include <cooperative_groups/reduce.h>
+#include <nvtx3/nvtx3.hpp>
+
+namespace mlx::core {
+namespace cu {
+
+namespace cg = cooperative_groups;
+
+template <int bits, int wsize = 8>
+inline constexpr __device__ short get_pack_factor() {
+  return (bits == 3 || bits == 5) ? 8 : (bits == 6 ? 4 : wsize / bits);
+}
+
+template <int bits, int wsize = 8>
+inline constexpr __device__ short get_bytes_per_pack() {
+  constexpr int power_of_2_bits = (bits & (bits - 1)) == 0;
+  return power_of_2_bits ? (wsize / 8) : (bits == 5 ? 5 : 3);
+}
+
+template <typename T, int group_size, int bits>
+__global__ void
+affine_quantize(const T* w, uint8_t* out, T* scales, T* biases, size_t size) {
+  auto block_size = cg::this_thread_block().dim_threads();
+  auto block_idx = cg::this_thread_block().group_index();
+  auto idx_in_block = cg::this_thread_block().thread_index();
+
+  auto tidx = block_idx.x * block_size.x + idx_in_block.x;
+  auto tidy = block_idx.y * block_size.y + idx_in_block.y;
+
+  auto grid_dim = cg::this_grid().dim_threads();
```

Comment:
`dim_threads` is not available in CUDA 12.2:

```
/home/zhao_cheng/codes/mlx/mlx/backend/cuda/quantized.cu(39): error: class "cooperative_groups::__v1::grid_group" has no member "dim_threads"
    auto grid_dim = cg::this_grid().dim_threads();
                                    ^

/home/zhao_cheng/codes/mlx/mlx/backend/cuda/quantized.cu(156): error: class "cooperative_groups::__v1::grid_group" has no member "dim_threads"
    auto grid_dim = cg::this_grid().dim_threads();
```

It is also not documented somehow:
https://docs.nvidia.com/cuda/cuda-c-programming-guide/#grid-group